docs/cvl/cvl2/changes.md

```diff
@@ -0,0 +1,583 @@
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Method literals require `sig:`
+
+In some places in CVL, you can refer to a contract method by its name and
+argument types.  For example, you might write
+```cvl
+require f.selector == balanceOf(address).selector;
+```
+
+In this example, `balanceOf(address)` is a *method literal*.  In CVL 2,
+these methods literals must now start with `sig:`.  For example, the above
+would become:
+
+```cvl
+require f.selector == sig:balanceOf(address).selector;
+```
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Stricter ordering on method annotations
+
+In CVL 2, the order of the annotations must be visibility modifiers (`internal` or `external`),
+followed by `returns` clause (if any), followed by `optional`, `library`, or `envfree` in any order (if any),
+followed by a summary (if any).
+
+CVL 1 was less strict about the order.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Use of contract name instead of `using` variable
+
+In CVL 1, the only way to refer to a contract on the scene was to first
+introduce a variable with a `using` statement, and then use that variable.  For
+example, to access a struct type `S` defined in `Example.sol`, you would need
+to write
+
+```cvl
+using Example as c;
+
+rule example {
+    c.S x = getAnS();
+}
+```
+
+In CVL 2, you must now use the name of the contract, rather than the variable,
+when referring to user-defined types.  The above example would now be written
+
+```cvl
+rule example {
+    Example.S x = getAnS();
+}
+```
+
+There is no need for a `using` statement in this example.
+
+`using` statements are still required to call methods on secondary contracts.
+
+```{todo}
+Error message
+```
+
+Changes to methods block entries
+--------------------------------
+
+In addition to the superficial changes listed above, there are some changes that
+change the way that methods block entries can be written.  In CVL 1, `methods`
+block entries often had several different functions and meanings:
+
+ - They are used to indicate targets for summarization
+ - They are used to write generic specs that could apply to contracts with
+   missing methods
+ - They are used to declare targets `envfree`
+
+With these changes, these different uses are more explicit.
+
+(cvl2-visibility)=
+### Required `internal` or `external` annotation
+
+Every methods block entry must be marked either `internal` or `external`.  The
+annotation must come after the argument list and before the `returns` clause.
+
+If a function is declared `public` in Solidity, then the Solidity compiler
+creates an internal implementation method, and an external wrapper method that
+calls the internal implementation.  Therefore, you can summarize a `public`
+method by marking the summarization `internal`.
+
+If you want to summarize both the internal implementation and the external
```

Comment:
This may or may not be the place to clarify that if the summaries are different, the `external` summary declaration will (I _believe_) take precedence, though if the `external` declaration has no summary, the internal summary will still be applied.

There is probably a better way to describe this but hopefully you get the drift

Reply:
I added a more extensive description